docs/contributing/docs-and-blog-components.md

```diff
@@ -206,6 +206,90 @@ New docs and blog posts are added to the [docs](https://github.com/gatsbyjs/gats
 
 You can read more about writing in Markdown in the [Markdown syntax guide](/docs/mdx/markdown-syntax).
 
+### Frontmatter
+
+[Frontmatter](/docs/adding-markdown-pages/#frontmatter-for-metadata-in-markdown-files) is a set of key value pairs that are injected into when building Markdown and MDX files. While authoring new docs and blog posts, it may be helpful to understand what frontmatter is available to you and how Gatsby expects it to be used.
+
+#### General
+
+- `title` (string)
+
+  The title of the doc or blog post. Gatsby renders the value in `og:title`, `<title>` and `<h1>`.
+
+- `excerpt` (string)
+
+  The excerpt for the post. Gatsby renders the value in `description`, `og:description`, and `twitter:description`.
+
+#### Blog Posts
+
+- `date` (string)
+
+  The blog post's date. GraphQL converts the string to a Date object and renders it with the format `MMMM Do YYYY`.
+
+- `canonicalLink` (string)
+
+  The canonical link. Gatsby renders the value in `<link rel="canonical">`.
```

Comment:
Perhaps we can link to an explanation of what this is for those that don't know. Maybe this one? https://support.google.com/webmasters/answer/139066?hl=en